### PR TITLE
bpo-34532: Fixed exit code for py.exe list versions arg

### PR DESCRIPTION
**Pull request by bgerrity**
Previously, any of the list version args would cause the launcher to exit with the `RC_NO_PYTHON` error code. This change causes the list version args to exit successfully.

https://bugs.python.org/issue34532

### CODE REVIEW

File: PC/launcher.c [hunk 1]
```diff
@@ -1600,14 +1601,15 @@ process(int argc, wchar_t ** argv)
...
         if ((argc == 2) && 
-            (!wcsncmp(p, L"-0", wcslen(L"-0")) || /* Starts with -0 or --list */
-            !wcsncmp(p, L"--list", wcslen(L"--list"))))
+            (!_wcsicmp(p, L"-0") || !_wcsicmp(p, L"--list") || /* Version args. */
+            !_wcsicmp(p, L"-0p") || !_wcsicmp(p, L"--list-paths")))
         {
-            valid = show_python_list(argv); /* Check for -0 or --list FIRST */
+            show_python_list(argv);
+            version_call = TRUE;
+            executable = NULL;
```
Comment (zooba) [inline PC/launcher.c]: Why not just return immediately from here? There doesn't seem to be any cleanup that has to happen.

Comment (bgerrity) [reply PC/launcher.c]: I wrote it to imitate the behavior of -h where it'll search for any Python version and return `RC_NO_PYTHON` if nothing is found. If that's not desirable, could just bump the check up a level.

**Comment by bgerrity** [comment]
Moved return up as advised

**Review by zooba — APPROVED** [review]

**Comment by zooba** [comment]
Doesn't need to go to 3.6

**Pull request merged**